microcosm_flask/tests/conventions/test_csv.py

```diff
@@ -0,0 +1,106 @@
+"""
+CRUD convention tests.
+
+"""
+from csv import reader
+from six import StringIO
+
+from enum import Enum
+
+from hamcrest import (
+    assert_that,
+    contains,
+    equal_to,
+    has_length,
+    is_,
+)
+
+from marshmallow.fields import String
+
+from microcosm.api import create_object_graph
+from microcosm_flask.conventions.csv_convention import configure_csv
+from microcosm_flask.namespaces import Namespace
+from microcosm_flask.operations import Operation
+from microcosm_flask.paging import OffsetLimitPageSchema
+from microcosm_flask.fields import QueryStringList, EnumField
+from microcosm_flask.tests.conventions.fixtures import (
+    Address,
+    AddressCSVSchema,
+    address_search,
+    person_search,
+    Person,
+    PersonCSVSchema,
+    PERSON_ID_1,
+)
+
+
+class TestEnum(Enum):
+    A = "A"
+    B = "B"
+
+    def __str__(self):
+        return self.value
+
+
+class SearchAddressPageSchema(OffsetLimitPageSchema):
+    list_param = QueryStringList(String())
+    enum_param = EnumField(TestEnum)
+
+
+def add_request_id(headers):
+    headers["X-Request-Id"] = "request-id"
+
+
+PERSON_MAPPINGS = {
+    Operation.Search: (person_search, OffsetLimitPageSchema(), PersonCSVSchema()),
+}
+
+
+ADDRESS_MAPPINGS = {
+    Operation.Search: (address_search, SearchAddressPageSchema(), AddressCSVSchema()),
+}
+
+
+class TestCSV(object):
```

Comment:
object not required

Reply:
I thought we wanted compatibility with Python 2 and 3?

Reply:
Fair.